### ISSUE TITLE
[Question] Assigning the PartitionKey

### ISSUE THREAD

**Issue by afacanerman** [IEvangelist/azure-cosmos-dotnet-repository#26]
Hi 

I am trying to use this package configuring the custom partition key (/userIdentifier) instead of /Id, but I am getting an error from cosmos client as below
`System.ArgumentException: The requested partition key path '/id' does not match existing Container 'VideoElapsedTimeData' with partition key path '/userIdentifier' (Parameter 'PartitionKey')`

```
public class VideoElapsedTimeData : Item
    {
        [JsonProperty("userIdentifier")] public string UserIdentifier { get; set; }
        [JsonProperty("consumedTimeInMs")] public double ConsumedTimeInMs { get; set; }
        [JsonProperty("metadata")] public string Metadata { get; set; } = string.Empty;
        [JsonProperty("videoType")] public string VideoType { get; set; } = string.Empty;
        [JsonProperty("viewedPercentage")] public double ViewedPercentage { get; set; }
        [JsonProperty("trimInMs")] public double TrimInMs { get; set; }
        [JsonProperty("trimOutMs")] public double TrimOutMs { get; set; }
        [JsonProperty("timestamp")] public DateTimeOffset Timestamp { get; set; } = DateTimeOffset.UtcNow;

        [JsonProperty("videoId")] public string VideoId { get; set; }

        protected override string GetPartitionKeyValue() => UserIdentifier;
    }
```
My container partition key set as /userIdentifier 

![image](https://user-images.githubusercontent.com/972726/102246695-f2201d80-3ef6-11eb-81af-5f5b1f482f4b.png)

Probably I am missing something I would be so glad if you could help, please.

Thank you

**Comment by afacanerman** [comment]
I think I might have solved the problem with putting

 **[PartitionKeyPath("/userIdentifier")]**
    public class VideoElapsedTimeData : Item
    {


on top of the SubItem and deleting the container and create via insert

**Comment by IEvangelist** [comment]
Yes, the `PartionKeyPath` needs to be specified in conjunction with overriding the `GetPartionKeyValue()`. If that works, we can close this issue. I did add source documentation on this:

- [PartitionKeyPathAttribute](https://github.com/IEvangelist/azure-cosmos-dotnet-repository/blob/main/Microsoft.Azure.CosmosRepository/src/Attributes/PartitionKeyPathAttribute.cs#L9-L13)
- [Item.GetPartitionKeyValue()](https://github.com/IEvangelist/azure-cosmos-dotnet-repository/blob/main/Microsoft.Azure.CosmosRepository/src/Item.cs#L59-L62)

But I suppose we could add to the _README.md_, thoughts?

**Comment by afacanerman** [comment]
Yes, it's working now. I have managed to solve it only after focusing on the source documentation. Would be nice to have an example though. Regardless Great job, thanks.

**Comment by jzhouw** [comment]
hi,  I tried the interface IItem and implemented a concrete class like below, the CRUD works but I noticed PartitionKey field in the db like below screenshot and looking not correct.. not sure.  Is this the correct usage of Item with PartitionKey attribute (didn't find an example of using IItem)

![image](https://user-images.githubusercontent.com/1085984/109475028-28117e80-7ab0-11eb-8457-340eca2bde7a.png)

```
[PartitionKeyPath("/tenantId")]
    public class JobItem: IItem
    {
        [JsonProperty("id")]
        public string Id { get; set; } = Guid.NewGuid().ToString();

        [JsonProperty("type")]
        public string Type { get; set; }
        public PartitionKey PartitionKey
        {
            get
            {
                return new PartitionKey(TenantId);
            }
        }

        [JsonProperty("tenantId")]
        public int TenantId { get; set; }
        public string Input { get; set; }
        public double Progress { get; set; }
        public DateTime? TimeStarted { get; set; }
        public DateTime? TimeCompleted { get; set; }
        public string Output { get; set; }
        public string Error { get; set; }
    }
```

**Comment by IEvangelist** [comment]
There is no reason to implement your own `IItem`, in this case just use the provided `Item` which already exposes the functionality you're looking for and override the `Item.GetPartitionKeyValue()`.
